pkg/patterns/addon/pkg/status/kstatus.go

```diff
@@ -70,22 +75,66 @@ func (k *kstatusAggregator) BuildStatus(ctx context.Context, info *declarative.S
 				log.Info("resource status was nil")
 				statusMap[status.UnknownStatus] = true
 			}
+			abnormalTrueConds := getAbnormalTrueConditions(ctx, unstruct)
+			totalAbnormalTrueConditions = append(totalAbnormalTrueConditions, abnormalTrueConds...)
 		}
 
-		aggregatedPhase := string(aggregateStatus(statusMap))
-
-		if currentStatus.Phase != aggregatedPhase {
-			currentStatus.Phase = aggregatedPhase
+		// Summarize all the deployment manifests statuses to a single results.
+		aggregatedPhase := aggregateStatus(statusMap)
+		// Update the Conditions for the declarativeObject status.
+		if aggregatedPhase == status.CurrentStatus {
+			SetReady(&currentStatus, totalAbnormalTrueConditions)
+		} else {
+			SetInProgress(&currentStatus, totalAbnormalTrueConditions)
 		}
+		currentStatus.Phase = string(aggregatedPhase)
 	}
 
 	if err := utils.SetCommonStatus(info.Subject, currentStatus); err != nil {
 		return err
 	}
+	return nil
+}
+
+// getAbnormalTrueConditions calculates the abnormal-true conditions that best describe the current state of the deployment manifests.
+func getAbnormalTrueConditions(ctx context.Context, unstruct *unstructured.Unstructured) []status.Condition {
+	log := log.FromContext(ctx)
+
+	// Normalize the deployment manifest conditions.
+	// The augmented condition "type" should only be "Stalled" or "Reconciling".
+	if err := status.Augment(unstruct); err != nil {
+		log.WithValues("object", unstruct).Error(err, "unable to augment conditions")
+		return nil
+	}
 
+	// The default unstructured.Unstructured does not have a structured "status.conditions" fields.
+	obj, err := status.GetObjectWithConditions(unstruct.Object)
+	if err != nil {
+		log.WithValues("object", unstruct).Error(err, "unable to get conditions")
+		return nil
+	}
+
+	if len(obj.Status.Conditions) == 0 {
+		return nil
+	}
+
+	// kstatus appends the augmented conditions to the end.
```

Comment:
Is there another method we can call into that is easier to use?

Reply:
Unfortunately no. I read all `kstatus/status` code, and my gut feeling is that kstatus is trying to make the surfaced functions extremely simple and handles a huge amount of legacy and convention logic under the hood.